docs/design/2021-07-29-hidden-sysvars.md

```diff
@@ -0,0 +1,176 @@
+* [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+
+Currently we create hidden/undocumented settings that are only designed for internal PingCAP users. In an internal call we unanimously agreed that this is problematic for our customers.
+
+## Motivation or Background
+
+The proposal is to change all hidden system variables to visible. The proposed solution differs based on the category of why the variable was made invisible.
+
+This proposal also includes expected standards for documentation.
+
+There are currently ~20 hidden system variables:
+
+| Name | Usage | Category |
+| ------|-------|-------------|
+| tidb_enable_local_txn | Local/Global Transaction | #2 Experimental |
+| tidb_txn_scope | Parameters of local transaction | #4 Non-boolean Experimental |
+| tidb_txn_read_ts | "" | #4 Non-boolean Experimental |
+| tidb_enable_pipelined_window_function | ‘Pipelined window function’ feature | #1 Stable |
+| tidb_enable_change_multi_schema | ‘Change multi schema in one statement’ feature. | #3 In Development Feature |
+| tidb_enable_point_get_cache | ‘point get cache’ feature | #2 Experimental |
+| tidb_enable_alter_placement | placement rules in SQL feature | #3 In Development Feature |
+| tidb_enable_extended_stats | ‘extended stats’ feature | #2 Experimental |
+| tidb_partition_prune_mode | Is partition prune mode dynamic or static | #2 Experimental |
+| tidb_enable_async_commit | Support Async Commit PRD  | #1 Stable |
+| tidb_enable_1pc | "" | #1 Stable |
+| tidb_guarantee_linearizability | "" | #1 Stable |
+| tidb_enable_index_merge_join | Index merge join | #2 Experimental |
+| tidb_enable_global_temporary_table | Global temporary table | #1 Stable |
+| tidb_enable_stable_result_mode | Make resultset sorted(deterministic order) by default | #1 Stable |
+| tidb_track_aggregate_memory_usage | Track memory usage of aggregate executor | #1 Stable |
+| tidb_enable_top_sql | Top SQL (Sprint3) 总设计文档  | #2 Experimental |
+| tidb_top_sql_agent_address | Parameters for Top SQL | #4 Non-boolean Experimental |
+| tidb_top_sql_precision_seconds | "" | #4 Non-boolean Experimental  |
+| tidb_top_sql_max_statement_count | "" | #4 Non-boolean Experimental |
+| tidb_top_sql_max_collect | "" | #4 Non-boolean Experimental |
+| tidb_top_sql_report_interval_seconds | "" | #4 Non-boolean Experimental |
+
+
+## Detailed Design
+
+### #1 “Stable” but intended to be deprecated or advanced configuration variables
+
+These variables were hidden because they are intended to be removed in a future version, and are not typically recommended to be changed.
+
+However; the problem with this decision is that these features do impact performance and behavior characteristics significantly, and can make debugging the difference between two clusters very difficult. This is exacerbated by our choice to only enable async commit for “new clusters”, and disable it for upgrades from earlier versions.
```

Comment:
> do impact performance and behavior characteristics significantly

I think this is a compatibility issue.

Reply:
> Does this affect the system variables hidden by SEM (https://github.com/pingcap/tidb/blob/master/docs/design/2021-03-09-security-enhanced-mode.md#system-variables)

It is not documented explicitly yet, but my intention is that SEM is not affected. The motivation for SEM is for security, but we could do a better job in future in documenting in `system-variables.md` which variables are affected.

Reply:
Yes it is. Usually the new scenarios will be faster for most cases, but there will be edge-cases where regressions exist. Not being able to observe differences in cluster configuration impacts debugging a lot.